File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/user_action/user_login/UserLoginActivity.java
package com.example.nightwingky.monstone.user_action.user_login;

import android.content.Intent;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.nightwingky.monstone.R;
import com.example.nightwingky.monstone.main_program.MainFragmentActivity;
import com.example.nightwingky.monstone.myConst.URLConst;
import com.example.nightwingky.monstone.user_action.user_forget_pwd.UserForgetPwdActivity;
import com.example.nightwingky.monstone.user_action.user_register.UserRegisterActivity;

import org.json.JSONException;

import java.io.IOException;

public class UserLoginActivity extends AppCompatActivity {

    private EditText edit_username;
    private EditText edit_password;

    private Button btn_login;
    private Button btn_forget_pwd;
    private Button btn_register;

    private static String userLoginURL = URLConst.getUser_login_url();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_layout_user_login);

        bindView();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    private void bindView() {

        edit_username = (EditText) findViewById(R.id.editText_login_username);
        edit_password = (EditText) findViewById(R.id.editText_login_pwd);
        btn_login = (Button) findViewById(R.id.btn_login_login);
        btn_register = (Button) findViewById(R.id.btn_login_register);
        btn_forget_pwd = (Button) findViewById(R.id.btn_login_forgetPwd);

        btn_login.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new UserLoginAsyncTask().execute(userLoginURL);
            }
        });

        btn_forget_pwd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(UserLoginActivity.this, UserForgetPwdActivity.class);
                startActivity(intent);
            }
        });

        btn_register.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(UserLoginActivity.this, UserRegisterActivity.class);
                startActivity(intent);
            }
        });
    }

    private String getJsonData() throws IOException, JSONException {
        String jsonString = UserLoginHttp.userLoginQuery(
                userLoginURL, edit_username.getText().toString(),
                edit_password.getText().toString());

        String success = UserLoginJsonConverter.convertJsonString(jsonString);

        return success;
    }

    class UserLoginAsyncTask extends AsyncTask<String, Void, String> {

        @Override
        protected String doInBackground(String... params) {

            try {
                return getJsonData();
            } catch (IOException e) {
                e.printStackTrace();
            } catch (JSONException e) {
                e.printStackTrace();
            }
            return "false";
        }

        @Override
        protected void onPostExecute(String s) {
            super.onPostExecute(s);

            if(s == "true") {
                Toast.makeText(UserLoginActivity.this, "登陆成功", Toast.LENGTH_SHORT).show();
                Intent intent = new Intent(UserLoginActivity.this, MainFragmentActivity.class);
                startActivity(intent);
                finish();
            } else {
                Toast.makeText(UserLoginActivity.this, "登录失败", Toast.LENGTH_SHORT).show();
            }
        }
    }

}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/user_action/user_login/UserLoginActivity.kt
package com.graduation.nightwingky.book_drifting.monstone.user_action.user_login

import android.content.Intent
import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast


import com.graduation.nightwingky.book_drifting.R
import com.graduation.nightwingky.book_drifting.monstone.main_program.MainFragmentActivity
import com.graduation.nightwingky.book_drifting.monstone.myConst.URLConst
import com.graduation.nightwingky.book_drifting.monstone.user_action.user_forget_pwd.UserForgetPwdActivity
import com.graduation.nightwingky.book_drifting.monstone.user_action.user_register.UserRegisterActivity
import kotlinx.android.synthetic.main.activity_layout_user_login.*

import org.json.JSONException

import java.io.IOException

class UserLoginActivity : AppCompatActivity() {

//    private var edit_username: EditText? = null
//    private var edit_password: EditText? = null
//
//    private var btn_login: Button? = null
//    private var btn_forget_pwd: Button? = null
//    private var btn_register: Button? = null

    private val jsonData: String
        @Throws(IOException::class, JSONException::class)
        get() {
            val jsonString = UserLoginHttp.userLoginQuery(
                    userLoginURL, mEdtLoginUsername!!.text.toString(),
                    mEdtLoginPassword!!.text.toString())

            return jsonString?.let { UserLoginJsonConverter.convertJsonString(it) }.toString()
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_layout_user_login)

        bindView()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    private fun bindView() {

//        edit_username = findViewById<View>(R.id.mEdtLoginUsername) as EditText
//        edit_password = findViewById<View>(R.id.mEdtLoginPassword) as EditText
//        btn_login = findViewById<View>(R.id.mBtnUserLogin) as Button
//        btn_register = findViewById<View>(R.id.mBtnUserRegister) as Button
//        btn_forget_pwd = findViewById<View>(R.id.mBtnUserForgetPwd) as Button

        mBtnUserLogin!!.setOnClickListener {
//            UserLoginAsyncTask().execute(userLoginURL)
            Toast.makeText(this@UserLoginActivity, "登陆成功", Toast.LENGTH_SHORT).show()
            val intent = Intent(this@UserLoginActivity, MainFragmentActivity::class.java)
            startActivity(intent)
            finish()
        }

        mBtnUserForgetPwd!!.setOnClickListener {
            val intent = Intent(this@UserLoginActivity, UserForgetPwdActivity::class.java)
            startActivity(intent)
        }

        mBtnUserRegister!!.setOnClickListener {
            val intent = Intent(this@UserLoginActivity, UserRegisterActivity::class.java)
            startActivity(intent)
        }
    }

    internal inner class UserLoginAsyncTask : AsyncTask<String, Void, String>() {

        override fun doInBackground(vararg params: String): String {

            try {
                return jsonData
            } catch (e: IOException) {
                e.printStackTrace()
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return "false"
        }

        override fun onPostExecute(s: String) {
            super.onPostExecute(s)

            if (s === "true") {
                Toast.makeText(this@UserLoginActivity, "登陆成功", Toast.LENGTH_SHORT).show()
                val intent = Intent(this@UserLoginActivity, MainFragmentActivity::class.java)
                startActivity(intent)
                finish()
            } else {
                Toast.makeText(this@UserLoginActivity, "登录失败", Toast.LENGTH_SHORT).show()
            }
        }
    }

    companion object {
        private val userLoginURL = URLConst.user_login_url
    }

}
